src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,389 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.indices;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.ProvisioningProgress;
+import org.opensearch.flowframework.model.State;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.model.WorkflowState;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.core.rest.RestStatus.INTERNAL_SERVER_ERROR;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX_MAPPING;
+import static org.opensearch.flowframework.common.CommonValue.META;
+import static org.opensearch.flowframework.common.CommonValue.NO_SCHEMA_VERSION;
+import static org.opensearch.flowframework.common.CommonValue.SCHEMA_VERSION_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_STATE_INDEX;
+
+/**
+ * A handler for global context related operations
+ */
+public class FlowFrameworkIndicesHandler {
+    private static final Logger logger = LogManager.getLogger(FlowFrameworkIndicesHandler.class);
+    private final Client client;
+    ClusterService clusterService;
+    private static final Map<String, AtomicBoolean> indexMappingUpdated = new HashMap<>();
+    private static final Map<String, Object> indexSettings = Map.of("index.auto_expand_replicas", "0-1");
```

Comment:
For understanding, why this index setting?

Reply:
This means the shard replica for our system index wont be set to some arbitrary number. This setting will allow automatically expending replica count to 1 if cluster needs it. Not 100% sure on the best numbers here, we can continue to adjust when doing more of a load testing.

Reply:
I am aware of the setting. Not sure why we are not letting OS handle it?

Reply:
I am not 100% sure what the default is by OS core. AD has 1-20 on job index. ML-commons has 0-1 on all indices, observability has 0-1 on all system indices. Mostly following ml-commons here, I can try to find more why it was put when it wasn't before.

Reply:
Let's address this in followup PR.